packages/lesswrong/server/scripts/mergeAccounts.js

```diff
@@ -0,0 +1,149 @@
+    collection,
+    documentId,
+    set: {[`${fieldName}.userId`]: targetUserId},
+    unset: {},
+    validate: false
+  })
+  // Update the revisions themselves
+  Revisions.update({ documentId, fieldName }, {$set: {userId: targetUserId}}, { multi: true })
+}
+
+const mergeReadStatusForPost = ({sourceUserId, targetUserId, postId}) => {
+  const sourceUserStatus = ReadStatuses.findOne({userId: sourceUserId, postId})
+  const targetUserStatus = ReadStatuses.findOne({userId: targetUserId, postId})
+  const readStatus = !!(sourceUserStatus?.isRead || targetUserStatus?.isRead)
+  const lastUpdated = new Date(sourceUserStatus?.lastUpdated) > new Date(targetUserStatus?.lastUpdated) ? sourceUserStatus?.lastUpdated : targetUserStatus?.lastUpdated
+  if (targetUserStatus) {
+    ReadStatuses.update({_id: targetUserStatus._id}, {$set: {isRead: readStatus, lastUpdated}})
+  } else if (sourceUserStatus) {
+    // eslint-disable-next-line no-unused-vars
+    const {_id, ...sourceUserStatusWithoutId} = sourceUserStatus
+    ReadStatuses.insert({...sourceUserStatusWithoutId, userId: targetUserId})
+  }
+}
+
+Vulcan.mergeAccounts = async (sourceUserId, targetUserId) => {
+  const sourceUser = Users.findOne({_id: sourceUserId})
+  const targetUser = Users.findOne({_id: targetUserId})
+
+  // Transfer posts
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Posts"})
+
+  // Transfer comments
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Comments"})
+
+  // Transfer conversations
+  await Conversations.update({participantIds: sourceUserId}, {$set: {"participantIds.$": targetUserId}}, { multi: true })
+
+  // Transfer private messages
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Messages"})
+
+  // Transfer notifications
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Notifications"})
+
+  // Transfer readStatuses
+  const readStatuses = await ReadStatuses.find({userId: sourceUserId}).fetch()
+  const readPostIds = readStatuses.map((status) => status.postId)
+  readPostIds.forEach((postId) => {
+    mergeReadStatusForPost({sourceUserId, targetUserId, postId})
+  })
+
+  // Transfer sequences
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Sequences"})
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Collections"})
+  
+  // Transfer karma
+  // eslint-disable-next-line no-console
+  console.log("Transferring karma")
+  await editMutation({
+    collection: Users,
+    documentId: targetUserId,
+    set: {karma: sourceUser.karma + targetUser.karma, afKarma: sourceUser.afKarma + targetUser.afKarma},
+    validate: false
+  })
+  
+  // Transfer votes that target content from source user (authorId)
+  // eslint-disable-next-line no-console
+  console.log("Transferring votes that target source user")
+  await Votes.update({authorId: sourceUserId}, {$set: {authorId: targetUserId}}, {multi: true})
```

Comment:
If both accounts have voted on the same post/comment, this isn't right. But it's probably okay to ignore the problem.